File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        int count = 0;
        vector<int> result = {};
        for (int i = 0; i<= n;i++){
            int j = i;
            while (j > 0){
                if (j % 2 == 1){
                    count +=1;    
                }
                j= j/2;
                
            }
            if (j == 0){
                result.push_back(count);
                count = 0;
                continue;
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = 0

        subarry = defaultdict(int)
    
        max_length = 0
        count = 0

        for right in range(len(s)):

            subarry[s[right]] += 1

            while (right-left+1 - max(subarry.values())) > k:
                subarry[s[left]] -= 1
                left+=1
                
            max_length=max(max_length, right - left + 1)   

        return max_length     

